### PR TITLE
Fix join button to team swaps

### PR DESCRIPTION
**Pull request by clayboone**
The problem behind this one is a complicated question of what happens when a user clicks the join button, then an admin changes or removes their role, then they click the button again.

If the user has no roles (or both roles), then their previous role from the database is assumed. If they have the opposite role (ie. an admin switched them from Knight to Viking), then the new role is assumed and our DB is updated.

### CODE REVIEW

File: pombot/lib/event_handlers.py [hunk 1]
```diff
@@ -38,68 +49,64 @@ async def on_raw_reaction_add_handler(bot: Bot, payload: RawReactionActionEvent)
...
-    }
+    if channel.name != Pomwars.JOIN_CHANNEL_NAME:
+        return
 
-    if (channel.name == Pomwars.JOIN_CHANNEL_NAME and payload.emoji.name in timezones):
+    if payload.emoji.name == Reactions.WAR_JOIN_REACTION:
+        await _create_roles_on_guild(bot_roles, guild)
+
+        team = _get_guild_team_or_random(payload.guild_id)
+        dm_description = "Enjoy to the Pom War event! Good luck and have fun!"
```
Comment (debashisbiswas) [inline pombot/lib/event_handlers.py:67]: Small typo here

File: pombot/lib/event_handlers.py [hunk 2]
```diff
@@ -38,68 +49,64 @@ async def on_raw_reaction_add_handler(bot: Bot, payload: RawReactionActionEvent)
...
+            user_roles = [r.name for r in payload.member.roles]
+            bot_roles_on_user = []
+
+            for bot_role in bot_roles:
+                try:
+                    bot_roles_on_user.append(user_roles[user_roles.index(bot_role)])
+                except ValueError:
+                    continue
+
+            if len(bot_roles_on_user) in [0, 2]:
```
Comment (debashisbiswas) [inline pombot/lib/event_handlers.py:82]: In the case the the user has both roles, should one of them be removed after sorting out which team they're actually on?

**Review by debashisbiswas — COMMENTED** [review]
My main comment was just a question I had while reading through the code - it's possible you've already handled that case and I didn't see it.

File: pombot/lib/event_handlers.py (hunk 2 above)
Comment (clayboone) [reply pombot/lib/event_handlers.py:82]: I didn't know what to do here, but there might be some testing an admin wants to do. If it doesn't affect the operation of the bot, I don't see why they can't have both roles.

Comment (debashisbiswas) [reply pombot/lib/event_handlers.py:82]: It does mean that those users would be able to see both channels, but honestly it won't be common to see users with two roles in the first place

**Pull request merged**